src/Models/BaseElement.php

```diff
@@ -293,6 +293,15 @@ public function canCreate($member = null, $context = array())
         return (Permission::check('CMS_ACCESS', 'any', $member)) ? true : null;
     }
 
+    public function write($showDebug = false, $forceInsert = false, $forceWrite = false, $writeComponents = false)
+    {
+        // Skips writes for broken blocks, so that we can still publish the page to allow all other blocks to publish.
+        if ($this->ObsoleteClassName) {
+            return $this->ID;
+        }
+        return parent::write($showDebug, $forceInsert, $forceWrite, $writeComponents);
+    }
+
```

Comment:
Could you test this method with broken and no broken elements?

Reply:
I'm not sure how (or if) I can get a broken element in a unit test context...